pkg/ircslack/channel.go

```diff
@@ -0,0 +1,90 @@
+package ircslack
+
+import (
+	"strings"
+
+	"github.com/slack-go/slack"
+)
+
+// Constants for public, private, and multi-party conversation prefixes.
+// Channel threads are prefixed with "+" but they are not conversation types
+// so they do not belong here. A thread is just a message whose destination
+// is within another message in a public, private, or multi-party conversation.
+const (
+	ChannelPrefixPublicChannel  = "#"
+	ChannelPrefixPrivateChannel = "@"
+	ChannelPrefixMpIM           = "&"
+	// NOTE: a thread is not a channel type
+	ChannelPrefixThread = "+"
+)
+
+// HasChannelPrefix returns true if the channel name starts with one of the
+// supproted channel prefixes.
+func HasChannelPrefix(name string) bool {
+	if len(name) == 0 {
+		return false
+	}
+	switch string(name[0]) {
+	case ChannelPrefixPublicChannel, ChannelPrefixPrivateChannel, ChannelPrefixMpIM, ChannelPrefixThread:
+		return true
+	default:
+		return false
+	}
+}
+
+// StripChannelPrefix returns a channel name without its channel prefix. If no
+// channel prefix is present, the string is returned unchanged.
+func StripChannelPrefix(name string) string {
+	if HasChannelPrefix(name) {
+		return name[1:]
+	}
+	return name
+}
+
+// Channel wraps a Slack conversation with a few utility functions.
+type Channel slack.Channel
+
+// IsPublicChannel returns true if the channel is public.
+func (c *Channel) IsPublicChannel() bool {
+	return c.IsChannel && !c.IsPrivate
+}
+
+// IsPrivateChannel returns true if the channel is private.
+func (c *Channel) IsPrivateChannel() bool {
+	return c.IsGroup && c.IsPrivate
+}
+
+// IsMP returns true if it is a multi-party conversation.
+func (c *Channel) IsMP() bool {
+	return c.IsMpIM
+}
+
+// IRCName returns the channel name as it would appear on IRC.
+// Examples:
+// * #channel for public groups
+// * @channel for private groups
+// * &Gxxxx|nick1-nick2-nick3 for multi-party IMs
+func (c *Channel) IRCName() string {
+	switch {
+	case c.IsPublicChannel():
+		return ChannelPrefixPublicChannel + c.Name
+	case c.IsGroup && c.IsPrivate:
```

Comment:
seems like  `c.IsPrivateChannel()` can be used here

Reply:
good catch, thanks!

Reply:
I'm still fixing a bunch of other things before this is ready to review, btw I've addressed your comment for the next update